internal/slave/service.go

```diff
@@ -20,38 +25,61 @@ type DKVService interface {
 	serverpb.DKVServer
 }
 
-type slaveService struct {
-	store       storage.KVStore
-	ca          storage.ChangeApplier
-	lg          *zap.Logger
-	statsCli    stats.Client
-	replCli     *ctl.DKVClient
-	replTckr    *time.Ticker
-	replStop    chan struct{}
-	replLag     uint64
-	fromChngNum uint64
+type ReplicationConfig struct {
+	// Max num changes to poll from master in a single replication call
 	maxNumChngs uint32
+	// Interval to periodically poll changes from master
+	replPollInterval time.Duration
+	// Maximum allowed replication lag from master for the slave to be considered valid
+	maxActiveReplLag uint64
+	// Maximum allowed replication time elapsed from master for the slave to be considered valid
+	// Applicable when replication requests are erroring out due to an issue with master / slave
+	maxActiveReplElapsed uint64
+	// Listener address of the master node
+	replMasterAddr string
 }
 
-// TODO: check if this needs to be exposed as a flag
-const maxNumChangesRepl = 10000
+func NewReplicationConfig(maxNumChanges uint32, replPollInterval time.Duration,
+	maxActiveReplLag uint64, maxActiveReplElapsed uint64) *ReplicationConfig {
+	return &ReplicationConfig{maxNumChngs: maxNumChanges, replPollInterval: replPollInterval,
+		maxActiveReplLag: maxActiveReplLag, maxActiveReplElapsed: maxActiveReplElapsed}
+}
```

Comment:
This function doesn't do anything apart from init of `ReplicationConfig`. In that case why can't we directly init and use`ReplicationConfig`

Reply:
Fixed in #81